helix-loader/src/lib.rs

```diff
@@ -177,6 +181,102 @@ pub fn merge_toml_values(left: toml::Value, right: toml::Value, merge_depth: usi
     }
 }
 
+/// This trait allows theme and icon flavors to be loaded from TOML files, with inheritance
+pub trait FlavorLoader<T> {
```

Comment:
I don't think a normal set of functions is a better fit here. I did a quick implementation to see how it turns out, and it is quite messy (functions with 8 arguments, one being an `impl Fn(toml::Value, toml::Value) -> toml::Value` to have access to the current `merge_flavors` function in `load_flavor`...). I could create structures to lower the number of arguments to pass but that would make the code unnecessarily complex IMO.
I think a trait is better suited here because `Icons` and `Themes` both have a shared behavior for the loading part, whilst implementing some of its aspects differently. Some of these functions are interdependent, which makes it messy outside of a trait. The trait implementation is also much more readable and much easier to use for anyone who want to implement a feature that needs this kind of loading.